src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -339,12 +339,55 @@ else if (!period.isEmpty()) {
                 .findFirst()
                 .map(SummaryJavadocCheck::getInlineTagNodeWithinHtmlElement);
 
-        if (node.isPresent() && !isSummaryTag(node.get())) {
+        if (node.isPresent() && (!isSummaryTag(node.get()) || !isDefinedFirst(node.get()))) {
             node = Optional.empty();
         }
         return node;
     }
 
+    /**
+     * Whether the {@code {@summary}} tag is defined first in the javadoc.
+     *
+     * @param inlineSummaryTag node of type {@link JavadocTokenTypes#JAVADOC_INLINE_TAG}
+     * @return {@code true} if the {@code {@summary}} tag is defined first in the javadoc
+     */
+    private static boolean isDefinedFirst(DetailNode inlineSummaryTag) {
+        boolean isDefinedFirst = true;
+        DetailNode previousSibling = JavadocUtil.getPreviousSibling(inlineSummaryTag);
+        while (previousSibling != null && isDefinedFirst) {
+            final int siblingType = previousSibling.getType();
+            if (siblingType == JavadocTokenTypes.TEXT) {
+                final String previousSiblingText = previousSibling.getText();
+                if (previousSiblingText.contains(DEFAULT_PERIOD)) {
+                    isDefinedFirst = false;
+                }
+            }
+            else if (siblingType == JavadocTokenTypes.HTML_ELEMENT) {
+                final DetailNode firstChild = JavadocUtil.getFirstChild(previousSibling);
+                final int firstChildType = firstChild.getType();
+                if (firstChildType == JavadocTokenTypes.PARAGRAPH
+                    || firstChildType == JavadocTokenTypes.HTML_TAG
+                    && isPreTag(firstChild)) {
```

Comment:
Why pre tag is so special? Any tag can be empty.

Reply:
```
/**
 * <pre>This is inside pre without period</pre>
 * {@summary This wont be included in summary}
 */
public class Test {
}
```
![Screenshot from 2022-03-13 10-09-11](https://user-images.githubusercontent.com/73882557/158045325-dd246bc7-2ac9-43e6-bca9-b671a5965508.png)

```
/**
 * <i>This is inside i tag with period.</i>
 * {@summary This would be inside the summary}
 */
public class Test {
}
```
![Screenshot from 2022-03-13 10-11-30](https://user-images.githubusercontent.com/73882557/158045382-e865a9d3-122d-4086-ba16-dd5608c75652.png)

There might be more tags like `pre`, I identified only `<p>` and `<pre>`.

Reply:
it is same as https://github.com/checkstyle/checkstyle/pull/11383#discussion_r826927226 , lets do violation on `<pre>This is inside pre without period</pre>` code.
if any text started javadoc before inlinetag, we consider it as summary. Any inlined tag after should not be considered.  

no violation on : `<i>This is inside i tag with period.</i>`

As you notice, I am ignoring inlined tag at all for such cases.

Reply:
I have filed a report, see comment above, what should be done with this PR till then, should go with our presumed bug (change implementation) or the current working?

Reply:
If at least few character(not html tags) placed before summary inlined tag We consider this character as summary and ignore inlined tag.

Reply:
Done